src/ctim/schemas/weakness.cljc

```diff
@@ -0,0 +1,202 @@
+  c/describable-entity-entries
+  c/sourcable-object-entries
+  (f/required-entries
+   (f/entry :type WeaknessTypeIdentifier
+            :description (str "The fixed value " type-identifier))
+   (f/entry :description c/Markdown
+            :description (str "should be short and limited to the key points "
+                              "that define this weakness"))
+   (f/entry :structure v/WeaknessStructure
+            :description (str "defines the structural nature of the weakness")))
+  (f/optional-entries
+   (f/entry :languages (f/set-of Language)
+            :description "Applicable Languages")
+   (f/entry :os (f/set-of OperatingSystem)
+            :description "Applicable operating systems")
+   (f/entry :architectures (f/set-of Architecture)
+            :description "Applicable architectures")
+   (f/entry :paradigms (f/set-of Paradigm)
+            :description "Applicable paradigms")
+   (f/entry :technologies (f/set-of Technology)
+            :description "Applicable technologies")
+   (f/entry :background_details c/Markdown
+            :description (str "information that is relevant but not related to "
+                              "the nature of the weakness itself"))
+   (f/entry :alternate_terms (f/set-of c/ShortString)
+            :description (str "indicates one or more other names used to "
+                              "describe this weakness"))
+   (f/entry :modes_of_introduction (f/set-of ModeOfIntroduction)
+            :description (str "information about how and when a given weakness "
+                              "may be introduced"))
+   (f/entry :likelihood v/HighMedLow
+            :description "Likelihood of exploit")
+   (f/entry :common_consequences (f/set-of Consequence)
+            :description (str "specify individual consequences associated "
+                              "with a weakness"))
+   (f/entry :detection_methods (f/set-of DetectionMethod)
+            :description (str "identify methods that may be employed to detect "
+                              "this weakness, including their strengths and "
+                              "limitations"))
+   (f/entry :potential_mitigations (f/set-of Mitigation)
+            :description (str "describe potential mitigations associated with "
+                              "a weakness"))
+   (f/entry :functional_areas (f/set-of v/FunctionalArea)
+            :description (str "identifies the functional area of the software "
+                              "in which the weakness is most likely to occur"))
+   (f/entry :affected_resources (f/set-of v/SystemResource)
+            :description (str "identify system resources that can be affected "
+                              "by an exploit of this weakness"))
+   (f/entry :notes c/Markdown
+            :description (str "provide any additional comments about the "
+                              "weakness"))))
+
+;; Does not contain these fields from CWE
+;; - Extended_Description
```

Comment:
Our Describable field allows us to add a longer 5000 character Markdown description, so we could potentially put a lot of these unincluded fields into that if we wished.  I think the ones you decided to omit seem reasonable.